### ISSUE TITLE
Large collections slow in runner UI

### ISSUE THREAD

**Issue by Mahonnant** [a85/POSTMan-Chrome-Extension#7201]
**Describe the bug**
Since the introduction of new UI for runner (adding the ability to select / reorder test cases), navigating a folders and subfolders of a large collection (2000+ requests organized with various folders/subfolders) is very slow.

**To Reproduce**
1. Launch runner with a large collection
2. Click on collection ==> wait 30+ seconds for the right panel to populate, folder list updates in left panel
3. Click on folder ==> wait 30+ seconds for the right panel to update, subfolder list updates in left panel
4. Click on subfolder ==> wait 30+ seconds for the right panel to update, subfolder list updates in left panel
5. ...

**Expected behavior**
Prior to the introduction of the selection / reorder features in right pane navigating folders / subfolders was very fast even with large collections.
Possible solutions : 
- populate the folder / subfolder list in the left pane first and don't freeze UI during right pane updating
- improve performances on right pane with large collections
- allow for deactivating right pane (I see the use case but I don't really use it)

**App information (please complete the following information):**
 - App Type : Native app
 - Postman Version : 7.5.0
 - OS: Window 10

**Comment by shamasis** [comment]
Indeed. We are tracking a performance slowdown of runner and are working on it.

**Comment by saswatds** [comment]
@Mahonnant We have made a lot of performance fixes on the runner front and they are available on our stable version of app `7.18.0`. Can you update to the latest version and check if the issue still persists for you?

**Comment by Mahonnant** [comment]
Hello Saswat,

Yes the issue still persists in v7.18.0.

I have a collection with 2000+ requests organized in different directories
and, when I select a directory in the runner it takes 30+ seconds to update
the screen.
In my opinion the right pane (the one where all requests are displayed) is
the one that takes time to process, this is quite frustrating because I
actually do not use the right pane at all (although I understand it could
be useful in some cases).

Regards,
Maël

**Comment by ramrochan** [comment]
At Postman, we are concentrating on Performance and recently a lot of Performance improvements have been made and we will be continuing the effort.

**Comment by danieladougan** [comment]
This happens to me as well. I'm on v9.18.2, which is the latest version. Whenever I try to run a large collection, I can get delays for HOURS sometimes before the run begins, or Postman will just crash completely.

What's odd is that my colleagues running the same collections are not experiencing anything like this.

**Comment by ghost** [comment]
Same issue here, Postman v10.9.0.

Collections from about 700~1000 items start to lag especially when using a single line/text output to console parsing a JSON response, with each request taking between 400-700ms. It will take some minutes to run, then take another few until the console outputs finish. The longer the collection, longer I have to wait at the end.

Options on the runner: Keep variable values, Save cookies after collection run.

Test on the request (one of my uses cases): 
```
if (responseBody !== "" && responseBody != undefined) {
    const jsonData = JSON.parse(responseBody);
    console.info(jsonData.balance.available.amount);
}
```

Also tried using "Clear cache and reload" mid execution but doesn't seem to help, though it seems to help if I clear the cache in between runs wth thousands of items.

**Comment by malvikach** [comment]
@georgerappelmeliuz thanks for the above details. 

I wanted to ask a few questions to get more information:

- Are you using Postman Desktop App or on the web?
- Do you face this while running the collection of (700-1000 requests) with multiple iterations? If yes, could you tell us more about the iterations your workflow needs.
- Have you tried using assertions to validate the tests instead of logging on console which seems to be a problem here. For example, asserting on jsonData.balance.available.amount  (I say this because, executing the collection doesn't seem to be the problem here but logging information on the console is adding some delays - we have had similar user reports in the past and while we understand that problem better, I want to see if this suggestion helps you) 

Please feel free to drop me a note if you like to meet on zoom to discuss the last point. 

Regards,
Malvika

**Comment by danieladougan** [comment]
I can't speak for @georgerappelmeliuz but for me it's the desktop app. There is only one iteration at a time when I run the collection. We have collections with thousands of tests. 

With regard to the console, sometimes I click "Clear All," and nothing happens because the logging has not yet caught up to the most recent test run. So, that might be a clue.

**Comment by Dhi13man** [comment]
Ran into this exact problem last year at work. We needed to run 1.5 million requests through an API for a data migration (credit card records moving between systems). Runner couldn't handle it. The file was too large to load into memory, execution was serial, and there was no way to track which requests failed mid-run.

I ended up building an open source tool to solve it: [Bombardment](https://bombardment.work) ([source](https://github.com/Dhi13man/bombardment-runner)). It streams records from CSV/JSON via Go channels (so memory stays flat regardless of file size), transforms them with JSONata expressions, and dispatches requests concurrently in batches across load-balanced targets. Every response gets tracked with status code, latency, and error message.

Not trying to pitch it as a Postman replacement (OBVIOUSLY). Postman is great for interactive API work. But for bulk execution at scale (thousands to millions of requests), the architecture needs to be fundamentally different: streaming instead of loading, concurrent instead of serial, CLI/server instead of Electron UI.

If anyone here is still hitting this wall, give it a try and let me know how it goes. Happy to help with setup.
